bootloader/src/pyi_utils.c

```diff
@@ -224,17 +224,21 @@ pyi_unsetenv(const char *variable)
 
 /* TODO rename fuction and revisit */
 int
-pyi_get_temp_path(char *buffer)
+pyi_get_temp_path(char *buffer, char *tempdir)
 {
     int i;
     wchar_t *wchar_ret;
     wchar_t prefix[16];
     wchar_t wchar_buffer[PATH_MAX];
 
-    /*
-     * Get path to Windows temporary directory.
-     */
-    GetTempPathW(PATH_MAX, wchar_buffer);
+    if (NULL != tempdir) {
+      wcscpy(wchar_buffer, tempdir);
```

Comment:
This should be a compiler error: Cannot implicitly convert `char *` to `wchar_t *`.

TOC entries are UTF-8 encoded on all platforms. tempdir must be converted to `wchar_t *` (UTF-16 on Windows) using `pyi_win32_utils_from_utf8()`